satellite/metainfo/metainfo.go

```diff
@@ -350,13 +350,28 @@ func (endpoint *Endpoint) createOrderLimitsForSegment(ctx context.Context, point
 	var combinedErrs error
 	var limits []*pb.AddressedOrderLimit
 	for _, piece := range pointer.GetRemote().GetRemotePieces() {
-		derivedPieceID := rootPieceID.Derive(piece.NodeId)
+		node, err := endpoint.cache.Get(ctx, piece.NodeId)
+		if err != nil {
+			endpoint.log.Debug("error getting node from overlay cache", zap.Error(err))
+			combinedErrs = errs.Combine(combinedErrs, err)
+			continue
+		}
+
+		if node != nil {
+			node.Type.DPanicOnInvalid("metainfo server order limits")
+		}
+
+		if !node.IsUp {
```

Comment:
maybe you have already kept an eye out for this, but I know Alex was running into a problem (pre-db merge) where the nodes in the overlay cache didn't have `IsUp` set to true, but they were implicitly Up by being in the overlay cache. I'm not totally sure where they would be set to up now but just something to look out for?

Reply:
or maybe this will be phased out by a new `lastContactSuccess`/`Failure` check..

Reply:
We have `lastContactSuccess/Failure` fields in the DBX model, but they are not exposed yet in the Golang/Protobuf structs. This will be done in the next PR that will introduce the NodeDossier struct that will have all the info from the DBX model.

So for this PR, I am just reusing the existing  `IsUp` field of `pb.Node` and set it properly in `convertDBNode` (overlaycache.go) based on the values of `lastContactSuccess/Failure`.

Reply:
cool, good to know!